### ISSUE TITLE
[tsp-client] TypeError: tspLocation.additionalDirectories.map is not a function

### ISSUE THREAD

**Issue by Alancere** [Azure/azure-sdk-tools#8551]
exec: `tsp-client update`
![image](https://github.com/Azure/azure-sdk-tools/assets/46921893/045d9f95-5471-4a70-a749-75f225403d13)

tsp-location.yaml 
from:
``` yaml
directory: specification/azurefleet/AzureFleet.Management
commit: 3cb1b51638616435470fc10ea00de92512186ece
repo: Azure/azure-rest-api-specs
additionalDirectories:
  - specification/mongocluster/DocumentDB.MongoCluster.Management
  - specification/apicenter/ApiCenter.Management
```
to ->
``` yaml
directory: specification/azurefleet/AzureFleet.Management
commit: 3cb1b51638616435470fc10ea00de92512186ece
repo: Azure/azure-rest-api-specs
additionalDirectories: specification/mongocluster/DocumentDB.MongoCluster.Management,specification/apicenter/ApiCenter.Management
```

### position
Change the tsp-location.yaml `additionalDirectories` in the following location: 
- https://github.com/Azure/azure-sdk-tools/blob/main/tools/tsp-client/src/index.ts#L321
- https://github.com/Azure/azure-sdk-tools/blob/main/tools/tsp-client/src/index.ts#L327

**Comment by catalinaperalta** [comment]
@Alancere can you provide more information, did the tool change the format of your tsp-location.yaml or did you do that manually?

**Comment by Alancere** [comment]
this my origin tsp-location.yaml content:
``` yaml
directory: specification/oracle/Oracle.Database
commit: 9a88bba565544d5907aaee7b10170fd770572904
repo: Azure/azure-rest-api-specs
additionalDirectories:
- specification/oracle/models
```

run `tsp-client update --commit 9a88bba565544d5907aaee7b10170fd770572904` result:
![image](https://github.com/user-attachments/assets/97144f91-ef1c-40df-a267-1e7e0e2c23d8)

tsp-location.yaml was changed by the tool to the following
``` yaml
directory: specification/oracle/Oracle.Database
commit: 9a88bba565544d5907aaee7b10170fd770572904
repo: Azure/azure-rest-api-specs
additionalDirectories: specification/oracle/models
```

**Comment by catalinaperalta** [comment]
Investigating, looks like a bug in the tool

**Comment by catalinaperalta** [comment]
@Alancere I have a fix, should be able to release a new version of tsp-client early next week.

**Comment by catalinaperalta** [comment]
FYI this was fixed in the v0.9.4 release. Please update to that version and give it a try
